app/models/orchestration_stack.rb

```diff
@@ -31,10 +31,7 @@ def tenant_identity
   end
 
   def directs_and_indirects(direct_attrs)
-    return send(direct_attrs) if descendants.blank?
-
-    indirects = descendants.inject([]) { |arr, child| arr << child.send(direct_attrs) }
-    (indirects << send(direct_attrs)).flatten
+    MiqPreloader.preload_and_map(subtree, direct_attrs)
```

Comment:
Looks like this is a great choice once we convert over to closure_trees. Will look at this when we make that conversion.

Holding off for now